### PR TITLE
#109 add log4j2

### PR DESCRIPTION
**Pull request by hossamgomaa24**

### CODE REVIEW

**Review by AshrakatAbdELSattar — COMMENTED** [review]
delete empty files and then add .log file in ignore

File: back-end/pom.xml [hunk 1]
```diff
@@ -63,8 +87,18 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-test</artifactId>
+			<exclusions>
+				<exclusion>
+					<groupId>org.springframework.boot</groupId>
+					<artifactId>spring-boot-starter-logging</artifactId>
```
[the remark this answers is not in the document]
Comment (hossamgomaa24) [reply back-end/pom.xml:93]: A conflict occur between "springboot logging" and "log4j2", it must be excluded from each dependancy.

File: back-end/logs/candi-meter.log [hunk 2]
```diff
@@ -0,0 +1,2 @@
+11-08-2022 14:31:39 ERROR DemoApplication:17 - Error level log message
+11-08-2022 14:37:11 ERROR DemoApplication:17 - Error level log message
```
[the remark this answers is not in the document]
Comment (hossamgomaa24) [reply back-end/logs/candi-meter.log]: the file was just for testing "log4j2" functionality, i deleted it after that.

File: back-end/src/main/resources/log4j2.Properties [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+appender.file.type=File
+appender.file.name = LOGFILE
+appender.file.fileName=${directoryname}/candi-meter.log
+appender.file.layout.type = PatternLayout
+appender.file.layout.pattern = %d{dd-MM-yyyy HH:mm:ss} %-5p %c{1}:%L - %m%n
+appender.file.filter.threshold.type = ThresholdFilter
+appender.file.filter.threshold.level = ERROR
+
+
```
[the remark this answers is not in the document]
Comment (hossamgomaa24) [reply back-end/src/main/resources/log4j2.Properties]: configuring "log4j2" with "properties" file has limitations, i switched to xml and deleted the "properties" file.

**Review by AshrakatAbdELSattar — APPROVED** [review]

**Pull request merged**